modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -155,159 +187,184 @@ PinMeshGenerator::PinMeshGenerator(const InputParameters & parameters)
   else
     _has_block_names = false;
 
-  // Define all id variables used in the pin
-  std::vector<unsigned int> ring_intervals;
-  std::vector<subdomain_id_type> ring_blk_ids;
-  std::vector<SubdomainName> ring_blk_names;
-  unsigned int background_intervals = 1;
-  std::vector<subdomain_id_type> background_blk_ids;
-  std::vector<SubdomainName> background_blk_names;
-  std::vector<unsigned int> duct_intervals;
-  std::vector<subdomain_id_type> duct_blk_ids;
-  std::vector<SubdomainName> duct_blk_names;
   // Initial block id used to define radial regions of pin
   unsigned int pin_block_id_start = 10000;
   // Use special block id to designate TRI elements
   unsigned int pin_block_id_tri = pin_block_id_start - 1;
 
-  for (const auto i : index_range(_intervals))
+  if (_homogenized)
   {
-    const auto block_name = "RGMB_PIN" + std::to_string(_pin_type) + "_R" + std::to_string(i);
-    const auto block_id = pin_block_id_start + i;
+    auto params = _app.getFactory().getValidParams("SimpleHexagonGenerator");
 
-    if (i < _ring_radii.size())
-    {
-      ring_intervals.push_back(_intervals[i]);
-      ring_blk_ids.push_back(block_id);
-      ring_blk_names.push_back(block_name);
-    }
-    else if (i > _ring_radii.size())
-    {
-      duct_intervals.push_back(_intervals[i]);
-      duct_blk_ids.push_back(block_id);
-      duct_blk_names.push_back(block_name);
-    }
-    else
-    {
-      background_intervals = _intervals[i];
-      background_blk_ids.push_back(block_id);
-      background_blk_names.push_back(block_name);
-    }
+    params.set<Real>("hexagon_size") = _pitch / 2.0;
+    params.set<boundary_id_type>("external_boundary_id") = 20000 + _pin_type;
+    const auto boundary_name = _is_assembly ? "outer_assembly_" + std::to_string(_pin_type)
+                                            : "outer_pin_" + std::to_string(_pin_type);
+    params.set<std::string>("external_boundary_name") = boundary_name;
+    params.set<subdomain_id_type>("block_id") =
+        _quad_center ? pin_block_id_start : pin_block_id_tri;
+    params.set<MooseEnum>("element_type") = _quad_center ? "QUAD" : "TRI";
+    auto block_name = "RGMB_PIN" + std::to_string(_pin_type) + "_R0";
```

Comment:
shouldnt this switch to assembly if is_assembly?

Reply:
This is just a placeholder value that tells the mesh generator which radial ring the mesh element is in during the `generate` stage. The actual block name value that the user sees is set in `generate` in the following line, where the name is set based on the value of `is_assembly`:

```
  const std::string default_block_name =
      std::string("RGMB_") + (_is_assembly ? std::string("ASSEMBLY_") : std::string("PIN_")) +
      std::to_string(_pin_type);
```

Reply:
sounds good